src/NuGet.Clients/NuGet.CommandLine/Commands/SearchCommand.cs

```diff
@@ -0,0 +1,174 @@
+
+namespace NuGet.CommandLine
+{
+
+    [Command(typeof(NuGetCommand), "search", "SearchCommandDescription",
+            UsageSummaryResourceName = "SearchCommandUsageSummary", UsageExampleResourceName = "SearchCommandUsageExamples")]
+    public class SearchCommand : Command
+    {
+        private readonly int _lineSeparatorLength = 20;
+
+        private readonly List<string> _sources = new List<string>();
+
+        [Option(typeof(NuGetCommand), "SearchCommandSourceDescription")]
+        public ICollection<string> Source => _sources;
+
+        [Option(typeof(NuGetCommand), "SearchCommandPreRelease")]
+        public bool PreRelease { get; set; } = false;
+
+        [Option(typeof(NuGetCommand), "SearchCommandTake")]
+        public int Take { get; set; } = 20;
+
+        private IList<PackageSource> GetEndpointsAsync()
+        {
+            List<PackageSource> configurationSources = SourceProvider.LoadPackageSources()
+                .Where(p => p.IsEnabled)
+                .ToList();
+
+            IList<PackageSource> packageSources;
+            if (Source.Count > 0)
+            {
+                packageSources = Source
+                    .Select(s => PackageSourceProviderExtensions.ResolveSource(configurationSources, s))
+                    .ToList();
+            }
+            else
+            {
+                packageSources = configurationSources;
+            }
+            return packageSources;
+        }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            string sourceSeparator = new string('=', _lineSeparatorLength);
+            string packageSeparator = new string('-', _lineSeparatorLength);
+
+            ILogger logger = Console;
+            CancellationToken cancellationToken = CancellationToken.None;
+
+            SearchFilter searchFilter = new SearchFilter(includePrerelease: PreRelease);
+
+            IList<PackageSource> listEndpoints = GetEndpointsAsync();
+
+            foreach (PackageSource source in listEndpoints)
+            {
+                SourceRepository repository = Repository.Factory.GetCoreV3(source);
+                PackageSearchResource resource = await repository.GetResourceAsync<PackageSearchResource>();
+
+                if (resource is null)
+                {
+                    continue;
+                }
+
+                IEnumerable<IPackageSearchMetadata> results = await resource.SearchAsync(
+                    string.Join(" ", Arguments).Trim(), // The arguments are joined with spaces to form a single query string
+                    searchFilter,
+                    skip: 0,
+                    take: Take,
+                    logger,
+                    cancellationToken);
+
+                Console.WriteLine(sourceSeparator);
+                Console.WriteLine($"Source: {source.Name}");
+
+                if (results.Any())
+                {
+                    if (Verbosity == Verbosity.Quiet)
+                    {
```

Comment:
Feels like we should not special case quiet. 

Maybe we write nothing to quiet or maybe the command itself can say quiet mode is not supported.
I've seen a couple of times and having to pick and choose between System.Console and our logger is just too error prone.

Reply:
In here, I'm looking for input from the whole team btw @NuGet/nuget-client :)

Reply:
We spoke about this offline, and agreed to push this to a future PR.